src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -75,12 +75,13 @@ internal abstract partial class HttpProtocol : IHttpResponseControl
         private Stream _requestStreamInternal;
         private Stream _responseStreamInternal;
 
-        public HttpProtocol(HttpConnectionContext context)
+        public void Initialize(HttpConnectionContext context)
         {
             _context = context;
 
             ServerOptions = ServiceContext.ServerOptions;
-            HttpRequestHeaders = new HttpRequestHeaders(reuseHeaderValues: !ServerOptions.DisableStringReuse);
+            HttpRequestHeaders.ReuseHeaderValues = !ServerOptions.DisableStringReuse;
```

Comment:
ServerOptions.DisableStringReuse can't change, right?

Reply:
I wasn't certain, hence I did this. It comes from KestrelServerOptions which I believe can't change.

Reply:
KestrelServerOptions can change, users have access to it, we just never told anybody what would happen if they did so.

Reply:
Incidentally, HttpSys does support live endpoint and cert changes. Kestrel could too but it would be a lot of work.